### PR TITLE
Feature/add image to post

### LINKED ISSUE #47: Re-Design the fragment_post.xml
you need to make the page inside ConstraintLayout  and make it scrolling

### PR DESCRIPTION
**Pull request by Hasa9nAli**
hello dears  reviewers this new branch to add new image  i work on 
1- change button to spinner 
2- get image from camera 
3 - get image from gallery 
4- add image to screen and cancel image from screen

### CODE REVIEW

File: presentation/ui/src/main/java/com/cupcake/ui/post/CreatePostFragment.kt [hunk 1]
```diff
@@ -1,33 +1,198 @@
...
+            checkPermissionOfCamera()
+        }
+        binding.buttonPhoto.setOnClickListener {
+            checkPermissionOfGallery()
+        }
+        setSpinnerAdapter()
+        lifecycleScope.launch {
+            viewModel.state.collect { state ->
+                if (state.isPostCreated) {
+                    Snackbar.make(requireView(), "Post Created Successfully", Snackbar.LENGTH_LONG)
```
Comment (AmeerAmjed) [inline presentation/ui/src/main/java/com/cupcake/ui/post/CreatePostFragment.kt]: why string hardCode

File: presentation/ui/src/main/java/com/cupcake/ui/post/CreatePostFragment.kt [hunk 2]
```diff
@@ -1,33 +1,198 @@
...
+                    binding.iconButtonRemoveImage.setOnClickListener {
+                        binding.relativeLayout.visibility = View.GONE
+                    }
+                }
+            }
+        }
+    }
+
+    private fun checkPermissionOfCamera() {
+        Dexter.withContext(this.requireContext()).withPermissions(
```
Comment (AmeerAmjed) [inline presentation/ui/src/main/java/com/cupcake/ui/post/CreatePostFragment.kt]: handle permissions with use plugin

File: presentation/ui/src/main/java/com/cupcake/ui/post/CreatePostFragment.kt [hunk 3]
```diff
@@ -1,33 +1,198 @@
...
+
+                override fun onNothingSelected(parent: AdapterView<*>?) {
+                    Toast.makeText(requireContext(), "Please Select the status visibility of post", Toast.LENGTH_SHORT).show()
+                }
+
+            }
+        }
+    }
+
+    companion object {
```
Comment (AmeerAmjed) [inline presentation/ui/src/main/java/com/cupcake/ui/post/CreatePostFragment.kt]: why **companion object**  not private

File: presentation/ui/src/main/res/drawable/cancal_icon.xml
Comment (AmeerAmjed) [inline presentation/ui/src/main/res/drawable/cancal_icon.xml:1]: why no following naming icon should name as ic_cancal

**Review by AmeerAmjed — CHANGES_REQUESTED** [review]

File: presentation/ui/src/main/java/com/cupcake/ui/post/CreatePostFragment.kt (hunk 1 above)
Comment (Hasa9nAli) [reply presentation/ui/src/main/java/com/cupcake/ui/post/CreatePostFragment.kt]: not me work on the this code this is for previous person but  i will fix it thank you for you notice

File: presentation/ui/src/main/java/com/cupcake/ui/post/CreatePostFragment.kt (hunk 2 above)
Comment (Hasa9nAli) [reply presentation/ui/src/main/java/com/cupcake/ui/post/CreatePostFragment.kt]: Done

File: presentation/ui/src/main/java/com/cupcake/ui/post/CreatePostFragment.kt [hunk 4]
```diff
@@ -11,23 +26,194 @@ import com.google.android.material.snackbar.Snackbar
...
-				}
-			}
-		}
-	}
+    override val LOG_TAG: String = this.javaClass.name
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        binding.buttonCamera.setOnClickListener {
```
Comment (TarekIdrees) [inline presentation/ui/src/main/java/com/cupcake/ui/post/CreatePostFragment.kt:37]: separate each button click in function

**Review by TarekIdrees — CHANGES_REQUESTED** [review]

**Pull request merged**